subsys/bluetooth/host/Kconfig

```diff
@@ -47,6 +48,26 @@ config BT_RX_BUF_LEN
 	  an L2CAP MTU of 65 bytes. On top of this there's the L2CAP header
 	  (4 bytes) and the ACL header (also 4 bytes) which yields 73 bytes.
 
+config BT_DISCARDABLE_POOL
```

Comment:
Why not use `BT_DISCARDABLE_BUF_COUNT != 0` instead of adding this Kconfig option?

Reply:
Or at the very least make this hidden and set it based on that

Reply:
> Why not use BT_DISCARDABLE_BUF_COUNT != 0 instead of adding this Kconfig option?

Mainly because it's more natural to have a boolean instead of indirect `> 0` in some places, but also because there are two levels of dependencies and it's (IMO) more natural to reference them with two options: first you have the drivers which are able to support the discardable pool (with this PR it's the native controller and H:4), so these are used to enable the pool in general. Next you have the second level, i.e. *if* the pool has been enabled what the default should be (e.g. a higher value if BT_MESH is enable). I find it easier to understand the dependencies with the split.

That said, if we forget about the slightly better readability of having a boolean the way to represent this with a single DISCARDABLE_BUF_COUNT option is to transform the DISCARDAB_POOL `default y` conditions into the DISCARDABLE_BUF_COUNT dependencies. That way it's not `!= 0` we'll use, rather `#if defined(CONFIG_BT_DISCARDABLE_BUF_COUNT)` which gets us back to a boolean again. Makes sense? It's *slightly* worse since we're then using an `int` type both as an int and as a general boolean (i.e. does the option exist at all or not).

Reply:
@carlescufi so if based on my explanation you're still in favour of a single option I'd do what I described above. One concrete difference is that you'll no-longer be able to force the option on unless the dependencies are fulfilled, but I guess that's desirable anyway, since it makes no sense to enable this unless we have a driver that can provide sufficiently nuanced information to the buffer allocators to take advantage of this feature.

Reply:
@carlescufi so the option definition would end up looking like this:
```
config BT_DISCARDABLE_BUF_COUNT
        int "Number of discardable event buffers"
        range 1 255
        default 20 if BT_MESH
        default 3
        depends on BT_H4 || BT_CTLR
        help
          Number of buffers in a separate buffer pool for events which
          the HCI driver considers discardable. Examples of such events
          could be e.g. Advertising Reports. The benefit of having such
          a pool means that the if there is a heavy inflow of such events
          it will not cause the allocation for other critical events to
          block and may even eliminate deadlocks in some cases.
```

I've not stretched the minimum to 0 since that makes no sense if we can simply either have this defined to something meaningful or not defined at all.

Reply:
@carlescufi I kind of like this single-option approach now that I tested implementing it - I've updated the PR accordingly.